### DISCUSSION TITLE
Add Bitbucket data center support

### DISCUSSION THREAD

**Discussion by anbraten** [woodpecker-ci/woodpecker#2364] [Request a Feature]
> It's kind of a bummer that support was removed, as [Bitbucket Data Center](https://www.atlassian.com/de/software/bitbucket/enterprise/data-center) (the on-premise license that is supported by Atlassian going forward) actually worked fine with the pre v1 versions.
>
> We are still actively using v0.14.3 with Bitbucket Data Center (a quite recent v8.8.2) and tried to migrate to next/v1 earlier this year (with https://github.com/woodpecker-ci/woodpecker/issues/1695 being the only blocker).

by @majodev  in https://github.com/woodpecker-ci/woodpecker/issues/1962#issuecomment-1702507127

**Comment by 6543** [comment]
I think we can add it back with somebody who do testing and provide use new updated test data cc @majodev

**Reply by mzampetakis** [reply]
Is there any way for someone to have access to a deployed Bitbucket Data Center or a license of it, for development and testing purposes?
I think this is the main limitation when someone has to develop the Bitbucket Data Center forge in woodpecker.

**Reply by majodev** [reply]
I'm not aware of any public offering and AFAIK evaluation licenses expire after ~1month.
We are currently hosting Bitbucket Data Center in k8s via this helm chart https://artifacthub.io/packages/helm/atlassian-data-center/bitbucket (which is using https://hub.docker.com/r/atlassian/bitbucket-server under the hood). Might be possible to at least automate setting up an instance for local usage (e.g. minikube/kind/k3s or even docker-compose) around that.

**Comment by langecode** [comment]
I have added PR #2503 which re-introduces an almost completely rewritten implementation for Bitbucket Server. We have a version of that branch running on-premises with around 10 projects with varying activity but it seems stable.

We are running Bitbucket Server v8.6.1 - so #1695 is no issue with the re-implemented Bitbucket support.

**Reply by langecode** [reply]
We are not running the "datacenter" version on-prem - only the standard Bitbucket server - so it may still be required to sanity check against a datacenter version - although the API should be the same.

**Comment by langecode** [comment]
I think actually it is only the "datacenter" version which is discontinued - the normal server edition has not been announced as end of life. At least that is what our Atlassian guy says - but I was also fooled by the announcement on the Atlassian website into thinking all on-prem Bitbucket would go away.

**Reply by mzampetakis** [reply]
I have [seen](https://www.atlassian.com/migration/assess/journey-to-cloud) about EOL for the bitbucket Server. I cannot find any reference about EOL for the "datacenter" version.

**Reply by anbraten** [reply]
That's what I saw.

![image](https://github.com/woodpecker-ci/woodpecker/assets/6918444/ee120281-be3f-409e-81a5-93464ed4221b)

**Comment by anbraten** [comment]
We now have Bitbucket Datacenter support with #2503
